packages/-ember-data/tests/integration/adapter/rest-adapter-test.js

```diff
@@ -1283,6 +1283,59 @@ module('integration/adapter/rest_adapter - REST Adapter', function (hooks) {
     assert.deepEqual(comments.toArray(), [comment1, comment2, comment3], 'The correct records are in the array');
   });
 
+  test('findHasMany - can push the smae record in twice and fetch the link', async function (assert) {
+    adapter.shouldBackgroundReloadRecord = () => false;
+    Post.reopen({ comments: DS.hasMany('comment', { async: true }) });
+
+    store.push({
+      data: {
+        type: 'post',
+        id: '1',
+        attributes: {
+          name: 'Rails is omakase',
+        },
+        relationships: {
+          comments: {
+            links: {
+              related: '/posts/1/comments',
+            },
+          },
+        },
+      },
+    });
+
+    store.push({
+      data: {
+        type: 'post',
+        id: '1',
+        attributes: {
+          name: 'Rails is omakase',
+        },
+        relationships: {
+          comments: {
+            links: {
+              related: '/posts/1/comments',
+            },
+          },
+        },
+      },
+    });
+
+    let post = store.peekRecord('post', '1');
+
+    ajaxResponse({
+      comments: [
+        { id: 1, name: 'FIRST' },
+        { id: 2, name: 'Rails is unagi' },
+        { id: 3, name: 'What is omakase?' },
+      ],
+    });
+
+    let comments = await post.get('comments');
+    assert.ok(post.comments.isFulfilled, 'comments promise is fulfilled');
```

Comment:
It looks like `post.comments` will trigger an [update of the promise proxy](https://github.com/sly7-7/data/blob/d328ef58c414123e96f8d7f9a4dcd49af15e1725/packages/store/addon/-private/system/model/internal-model.ts#L715). That's why this assertion fails. Removing that proxy update fixes this particular test, but obviously breaks a few others. 

I guess the question is what we expect to get from consecutive `post.comments` calls. Are they supposed to return the cached results or do they all trigger a new fetch?

Reply:
As I said [in a previous comment](https://github.com/emberjs/data/pull/7691#issuecomment-915358747) I don't know how calls of `post.comments` should behave :(. And it looks like that calling `post.comments` won't always trig the update you mentioned (since DS.hasMany() is a computed property).

Reply:
I should have read all the comments before diving into the code 😬 🙈